test/runnable/test18076.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+dir=${RESULTS_DIR}${SEP}runnable
+output_file=${dir}/test18076.sh.out
+
+echo 'import std.stdio; void main() { writeln("Success"); }' | \
+	$DMD -m${MODEL} -run - > ${output_file} || exit 1
+grep -q '^Success$' ${output_file} || exit 1
```

Comment:
FWIW the `|| exit` is superfluous here.
A bash script will yield the last return value as exit code if not explicitly exited

```bash
sh -c "grep -q 'foo' afile.d" ; echo $? 
```

Reply:
True.  I'm wary of relying on these implicit behaviours, though.  Stating `|| exit 1` makes it more obvious what's happening.  Not that it makes a difference, I just prefer to have things up-front than implicit. Implicit things tend to be fragile and unintentionally break later.

Reply:
That's why one usually does a `set -e` at the beginning: https://unix.stackexchange.com/questions/15998/what-does-the-e-do-in-a-bash-shebang :wink: